### ISSUE TITLE
Memory leaks occur when the grpc client connection fails

### ISSUE THREAD

**Issue by Chise1** [grpc/grpc#36117]
### What version of gRPC and what language are you using?

grpcio-1.62.1

protobuf-5.26.0

In fact, the first version I found this problem with was 1.57.1

### What operating system (Linux, Windows,...) and version?

WSL2  (This problem also exists in ubuntu22.04)

### What runtime / compiler are you using (e.g. python version or version of gcc)

python 3.10

### What did you do?
Please provide either 1) A unit test for reproducing the bug or 2) Specific steps for us to follow to reproduce the bug. If there’s not enough information to debug the problem, gRPC team may close the issue at their discretion. You’re welcome to re-open the issue once you have a reproduction.

When I try to connect to a server and fail, I find that the memory is not reduced by the failed connection. If I keep trying to connect, memory will gradually increase.

example code:
 

```python
import asyncio
import logging
import time

import grpc
from grpc.aio import AioRpcError

import helloworld_pb2
import helloworld_pb2_grpc


async def connect(address: str) -> None:
    try:
        async with grpc.aio.insecure_channel(f"{address}:50051") as channel:
            stub = helloworld_pb2_grpc.GreeterStub(channel)
            response = await stub.SayHello(helloworld_pb2.HelloRequest(name="you"))
    except AioRpcError as e:
        pass

async def run() -> None:
    while True:
        await connect("192.168.202.232")
        await connect("172.21.96.142")

if __name__ == "__main__":
    logging.basicConfig()
    asyncio.run(run())

```

I use docker run it.
dockerfile :

```Dockerfile

FROM python:3.10-slim
RUN mkdir /grpc-client
WORKDIR /grpc-client
COPY async_greeter_client.py /grpc-client/
COPY helloworld_pb2.py /grpc-client/
COPY helloworld_pb2.pyi /grpc-client/
COPY helloworld_pb2_grpc.py /grpc-client/

RUN pip3 install  grpcio
RUN pip3 install  protobuf
CMD ["python","async_greeter_client.py"]

```

### What did you expect to see?

After the connection fails, memory should be reduced, not unchanged.

### What did you see instead?

The memory does not decrease after the connection fails, and each time the ip is repeatedly connected, the memory increases a little.

### Anything else we should know about your project / environment?
no

**Comment by gnossen** [comment]
@Chise1 Can you confirm how much you saw the memory increase over time? Does the increase level off? Or does the memory increase monotonically over time?

**Comment by Chise1** [comment]
I use k8s to run python programs, and with prometheus I can see that memory usage increases a bit every once in a while.

**Comment by Chise1** [comment]
IN python3.13,fix unix socket.Is this the same problem that I'm talking about?

[asyncio.loop.create_unix_server()](https://docs.python.org/3.13/library/asyncio-eventloop.html#asyncio.loop.create_unix_server) will now automatically remove the Unix socket when the server is closed. (Contributed by Pierre Ossman in [gh-111246](https://github.com/python/cpython/issues/111246).)

**Comment by XuanWang-Amos** [comment]
Glad to hear that you found the cause! We'll close this one since it's a CPython issue.

**Comment by louis-jaris** [comment]
@Chise1 did you try to validate your hypothesis (bug in CPython, fixed in python 3.13) by changing the version in your `Dockerfile` ? If you did not / can't, could you please provide us with all the files requires to run your example please? 🙏 

Thanks to your exploration (and discovery that the first version of `grpcio` that have this memory issue is `1.57.1`), I was able to see the code delta that could have caused this issue: https://github.com/grpc/grpc/compare/v1.57.0...v1.57.1

---

I'd like to be sure at 100% if upgrading to python 3.13 is **a** solution.
And I'd would also be interested in seeing a fix in `grpcio`, as upgrading to higher python is not necessarily a solution for all python users (e.g. all the persons running on Amazon Linux 2023, exposing and maitaining only `python3.9` and `python3.11`, c.f. [all the packages available in AL2023](https://docs.aws.amazon.com/linux/al2023/release-notes/all-packages.html))

@XuanWang-Amos could you consider re-open this issue given the context above? Thank you 🙏
